src/routes/register_request.rs

```diff
@@ -139,3 +142,46 @@ pub(super) async fn handle_image(
     .await
     .unwrap()
 }
+
+pub(super) async fn handle_font(
+    State(api): State<ApiState>,
+    request: Request,
+) -> Result<Response, ApiError> {
+    let Some(content_type) = request.headers().get("Content-Type") else {
+        return Err(ApiError::malformed_request(&"Missing Content-Type header"));
+    };
+
+    if let Ok(content_type_str) = content_type.to_str() {
+        if !content_type_str.starts_with("multipart/form-data") {
+            return Err(ApiError::malformed_request(&"Invalid Content-Type"));
+        }
+    } else {
+        return Err(ApiError::malformed_request(&"Invalid Content-Type"));
+    }
+
+    let mut multipart = Multipart::from_request(request, &api)
+        .await
+        .map_err(|err| ApiError::malformed_request(&err))?;
+    let Some(field) = multipart
+        .next_field()
+        .await
+        .map_err(|err| ApiError::malformed_request(&err))?
+    else {
+        return Err(ApiError::malformed_request(&"Missing font file"));
+    };
+
+    let bytes = field
+        .bytes()
+        .await
+        .map_err(|err| ApiError::malformed_request(&err))?;
+
+    tokio::task::spawn_blocking(move || {
+        Pipeline::register_font(
+            &api.pipeline.lock().unwrap(),
+            Source::Binary(Arc::new(bytes.to_vec())),
```

Comment:
create binary outside of the task

Currently you are 
- crearating Bytes object
- converting it to Vec
- and creating the Arc<Vec<>> from that

I think you can create Source::Binary  without that copy